File: testapp-tomcat/src/test/java/com/example/MainViewTest.java
package com.example;

import com.github.mvysny.kaributesting.v10.MockVaadin;
import com.github.mvysny.kaributesting.v10.Routes;
import com.vaadin.flow.component.UI;
import org.jetbrains.annotations.NotNull;
import org.junit.jupiter.api.*;

import static com.github.mvysny.kaributesting.v10.LocatorJ._assertOne;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * Karibu tests. Quick tests without you having to actually start Jetty. PREFERRED WAY OF TESTING YOUR APPS.
 */
public class MainViewTest {
    @NotNull
    private static final Routes routes = new Routes().autoDiscoverViews("com.example");

    @BeforeAll
    public static void setupApp() {
        assertFalse(Bootstrap.initialized);
        new Bootstrap().contextInitialized(null);
    }

    @AfterAll
    public static void tearDownApp() {
        new Bootstrap().contextDestroyed(null);
        assertFalse(Bootstrap.initialized);
    }

    @BeforeEach
    public void setupVaadin() {
        MockVaadin.setup(routes);
    }

    @AfterEach
    public void tearDownVaadin() {
        MockVaadin.tearDown();
    }

    @Test
    public void smoke() {
        UI.getCurrent().navigate(MainView.class);
        _assertOne(MainView.class);
        assertTrue(Bootstrap.initialized);
    }
}

File: testapp-kotlin-tomcat/src/test/kotlin/com/example/MainViewTest.kt
package com.example

import com.github.mvysny.kaributesting.v10.LocatorJ
import com.github.mvysny.kaributesting.v10.MockVaadin.setup
import com.github.mvysny.kaributesting.v10.MockVaadin.tearDown
import com.github.mvysny.kaributesting.v10.Routes
import com.vaadin.flow.component.UI
import org.junit.jupiter.api.*

/**
 * Karibu tests. Quick tests without you having to actually start Jetty. PREFERRED WAY OF TESTING YOUR APPS.
 */
class MainViewTest {
    @BeforeEach
    fun setupVaadin() {
        setup(routes)
    }

    @AfterEach
    fun tearDownVaadin() {
        tearDown()
    }

    @Test
    fun smoke() {
        UI.getCurrent().navigate(
            MainView::class.java
        )
        LocatorJ._assertOne(MainView::class.java)
        Assertions.assertTrue(Bootstrap.initialized)
    }

    companion object {
        private val routes = Routes().autoDiscoverViews("com.example")
        @BeforeAll
        @JvmStatic
        fun setupApp() {
            Assertions.assertFalse(Bootstrap.initialized)
            Bootstrap().contextInitialized(null)
        }

        @AfterAll
        @JvmStatic
        fun tearDownApp() {
            Bootstrap().contextDestroyed(null)
            Assertions.assertFalse(Bootstrap.initialized)
        }
    }
}
